src/Compilers/CSharp/Portable/Symbols/PropertySymbolExtensions.cs

```diff
@@ -70,5 +70,13 @@ public static bool HasRefOrOutParameter(this PropertySymbol property)
             }
             return false;
         }
+
+        public static bool ShouldDisplayReadOnly(this IPropertySymbol property)
+        {
+            // If at least one accessor is present and all present accessors are readonly, the property should be marked readonly.
+            return (property.GetMethod != null || property.SetMethod != null) &&
+                property.GetMethod?.IsReadOnly != false &&
+                property.SetMethod?.IsReadOnly != false;
```

Comment:
can this just be `=> property.GetMethod?.IsReadOnly || property.SetMethod?.IsReadOnly`;?

Reply:
Only reason to do it this way is to prevent (malformed) properties with no accessors from being displayed as 'readonly'. Maybe this is such an edge case that it doesn't matter?

Reply:
ahh... i can see some compelxity there.  you want to handle the 'null' case as well as the 'true' case.  fair enough :)